src/govuk/i18n.unit.test.mjs

```diff
@@ -132,5 +132,156 @@ describe('I18n', () => {
       const i18n = new I18n(config)
       expect(i18n.t('nameString', { name: '%{name}' })).toBe('My name is %{name}')
     })
+
+    it('formats numbers that are passed as placeholders', () => {
+      const i18n = new I18n({
+        ageString: 'I am %{age} years old'
+      })
+
+      expect(i18n.t('ageString', { age: 2000 })).toBe('I am 2,000 years old')
+    })
+
+    it('does not format number-like strings that are passed as placeholders', () => {
+      const i18n = new I18n({
+        yearString: 'Happy new year %{year}'
+      })
+
+      expect(i18n.t('yearString', { year: '2023' })).toBe('Happy new year 2023')
```

Comment:
Do we need a test to ensure that the number formatting changes depending on `locale`?

Reply:
I considered it, but this struck me as more a test of the 'is this a number' logic rather than the number formatting functionality itself—as that functionality is being handled by the browser API—it didn't feel necessary to test more than once.

Reply:
My thinking is that it checks that we're passing `locale` to `Intl.NumberFormat` correctly – we're not somehow hardcoding it to English or passing `undefined` which would actually still pass this test (at least in Chrome):

```js
new Intl.NumberFormat(undefined).format(1000)
// -> '1,000'
```